### PR TITLE
Added `autoImplementLibrary` member to the `testify` extension.

### PR DESCRIPTION
**Pull request by DanielJette**
## What?

Added `autoImplementLibrary` member to the `testify` extension.
Defaults to `true`.
When set to `false`, prevents the Plugin from automatically adding a Library `androidTestImplementation` dependency to your project.
This is useful for local debugging or if you require a different version of the library and plugin.

## Why?

It's sometimes necessary to run a different version of the library than the plugin. A good example would be when developing locally and needing to test new library features against the existing plugin. This fix allows you to break the version dependency between the library and plugin and import either/or separately.

## How

Add a `autoImplementLibrary` property to the `testify` extension

## Tophat

- Check out this branch locally
- Run `./gradlew Sample:dependencies | grep testify`
    Note 0 results
- Edit the `build.gradle` file in the Sample app. 
    Remove `androidTestImplementation project(":Library")`
    To the bottom of the file, add the following:
    ```
    testify {
        autoImplementLibrary true
    }
    ```
- Run `./gradlew Sample:dependencies | grep testify` again.
   Note how the library is now a dependency

### CODE REVIEW

**Comment by fredells** [comment]
looking at travis:
https://travis-ci.com/github/Shopify/android-testify/builds/171875665
```
Time: 168.807
There was 1 failure:
1) setOrientation(com.shopify.testify.sample.ScreenshotRuleExampleTests)
com.shopify.testify.internal.exception.ScreenshotIsDifferentException: 
*  The captured screenshot is different from the baseline screenshot.
*  Run `./gradlew Sample::screenshotPull` to view the differences.
*  Run `./gradlew Sample::screenshotTest -PtestClass=com.shopify.testify.sample.ScreenshotRuleExampleTests#setOrientation` to run this test again.
```
does this just need to be rebased on the orientation changes?

**Comment by DanielJette** [comment]
> does this just need to be rebased on the orientation changes?

Yes, that's it exactly
Orientation change tests are flaky on CI here
Which is good because it helped me develop the fix. I'm rebasing now

File: Plugin/src/main/kotlin/com/shopify/testify/TestifyExtension.kt [hunk 1]
```diff
@@ -95,6 +96,9 @@ internal data class TestifySettings(
         if (extension.installAndroidTestTask?.isNotEmpty() == true) {
             this.installAndroidTestTask = extension.installAndroidTestTask
         }
+        if (extension.autoImplementLibrary != null) {
```
Comment (fredells) [inline Plugin/src/main/kotlin/com/shopify/testify/TestifyExtension.kt:99]: is this even possible?

Comment (fredells) [reply Plugin/src/main/kotlin/com/shopify/testify/TestifyExtension.kt:99]: for it to be `null` i mean

Comment (DanielJette) [reply Plugin/src/main/kotlin/com/shopify/testify/TestifyExtension.kt:99]: Yes, it's possible

There are two classes that work together in this code. The Setting and the Extension
The Extension is a gradle concept. It looks like `testify {}` in your build.gradle file.
In the Extension (which this line is checking) everything can be null. In this context, null means "not specified"

The Settings are a concept unique to Testify. It's a cache of the Extension values, fully resolved and with all defaults applied as necessary.

In this block, we're checking to see if the `autoImplementLibrary` extension is `not specified` (null) or some value.
If it's a some value, we override Settings to that.

**Pull request merged**